warehouse/source/http.go

```diff
@@ -0,0 +1,188 @@
+		switch strings.ToLower(tableName) {
+		case "rudder_discards", "rudder_identity_mappings", "rudder_identity_merge_rules":
+			return false
+		default:
+			return true
+		}
+	})
+
+	type metadata struct {
+		JobRunID  string `json:"job_run_id"`
+		TaskRunID string `json:"task_run_id"`
+		JobType   string `json:"jobtype"`
+		StartTime string `json:"start_time"`
+	}
+
+	metadataJson, err := json.Marshal(metadata{
+		JobRunID:  payload.JobRunID,
+		TaskRunID: payload.TaskRunID,
+		StartTime: payload.StartTime,
+		JobType:   string(notifier.JobTypeAsync),
+	})
+	if err != nil {
+		m.logger.Errorw("marshalling metadata for inserting source job", lf.Error, err.Error())
+		http.Error(w, "can't marshall metadata", http.StatusInternalServerError)
+		return
+	}
+
+	jobIds, err := m.sourceRepo.Insert(r.Context(), lo.Map(tableNames, func(item string, index int) model.SourceJob {
+		return model.SourceJob{
+			SourceID:      payload.SourceID,
+			DestinationID: payload.DestinationID,
+			WorkspaceID:   payload.WorkspaceID,
+			TableName:     item,
+			JobType:       payload.JobType,
+			Metadata:      metadataJson,
+		}
+	}))
+	if err != nil {
+		if errors.Is(r.Context().Err(), context.Canceled) {
+			http.Error(w, ierrors.ErrRequestCancelled.Error(), http.StatusBadRequest)
+			return
+		}
+		m.logger.Errorw("inserting source jobs", lf.Error, err.Error())
+		http.Error(w, "can't insert source jobs", http.StatusInternalServerError)
+		return
+	}
+
+	resBody, err := json.Marshal(insertJobResponse{
+		JobIds: jobIds,
+		Err:    nil,
+	})
+	if err != nil {
+		m.logger.Errorw("marshalling response for inserting source job", lf.Error, err.Error())
+		http.Error(w, ierrors.ErrMarshallResponse.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	_, _ = w.Write(resBody)
+}
+
+// StatusJobHandler The following handler gets called for getting the status of the async job
+func (m *Manager) StatusJobHandler(w http.ResponseWriter, r *http.Request) {
+	defer func() { _ = r.Body.Close() }()
+
+	queryParams := r.URL.Query()
+	payload := insertJobRequest{
+		TaskRunID:     queryParams.Get("task_run_id"),
+		JobRunID:      queryParams.Get("job_run_id"),
+		SourceID:      queryParams.Get("source_id"),
+		DestinationID: queryParams.Get("destination_id"),
+		WorkspaceID:   queryParams.Get("workspace_id"),
+	}
+	if err := validatePayload(&payload); err != nil {
+		m.logger.Warnw("invalid payload for source job status", lf.Error, err.Error())
+		http.Error(w, fmt.Sprintf("invalid request: %s", err.Error()), http.StatusBadRequest)
+		return
+	}
+
+	sourceJob, err := m.sourceRepo.GetByJobRunTaskRun(r.Context(), payload.JobRunID, payload.TaskRunID)
+	if err != nil {
+		if errors.Is(r.Context().Err(), context.Canceled) {
+			http.Error(w, ierrors.ErrRequestCancelled.Error(), http.StatusBadRequest)
+			return
+		}
+		if errors.Is(err, model.ErrSourcesJobNotFound) {
+			http.Error(w, model.ErrSourcesJobNotFound.Error(), http.StatusBadRequest)
```

Comment:
Not sure what's the contract here, just thought I'd mention that a 404 probably makes more sense.